### DISCUSSION TITLE
Trying to implement teams

### DISCUSSION THREAD

**Discussion by rabol** [spatie/laravel-permission#2432] [Q&A]
Hi

I'm trying to implement teams support in my SAAS app and is struggling a little.

I'm using: https://github.com/mpociot/teamwork for the teams functionality

I have implemented the Middleware, and I have the Gate:before for super admin.

as and example,I would like 3 roles Basic, Standard and Pro

If I create the roles I can see that the 'current_team_id is null, which I find okay as I do not know the team id yet as no team has been created.

If I then create a user and try to assign any role to that user I get the:

```
 SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'current_team_id' cannot be null (Connection: mysql, SQL: insert into `model_has_roles` (`current_team_id`, `model_id`, `model_type`, `role_id`) values (?, 1, App\Models\User, 1))
```

Sure at this point in time the user is simply a user, no teams has been created.

Does this package not allow both users with no team and uses that is part of a team ?

**Comment by parallels999** [comment]
Read other discussions
https://github.com/spatie/laravel-permission/discussions/2088#discussioncomment-5866696
